.semaphore/semaphore.yml

```diff
@@ -16,4 +16,43 @@ blocks:
       jobs:
         - name: bundle
           commands:
-          - checkout
\ No newline at end of file
+            - checkout
+            - cache restore gems-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),gems-$SEMAPHORE_GIT_BRANCH-,gems-master-
+            - cache restore node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),node-modules-$SEMAPHORE_GIT_BRANCH,node-modules-master
+            - cd /home/semaphore/.rbenv/plugins/ruby-build && git pull && cd -
+            - sem-version ruby $(cat .ruby-version)
+            - bundle install --deployment -j 4 --path vendor/bundle
+            - yarn install
+            - cache store gems-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock) vendor/bundle
+            - cache store node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock) node_modules
+
+  - name: Tests
+    task:
+      env_vars:
+        - name: RAILS_ENV
+          value: test
+        - name: DATABASE_URL
+          value: postgresql://postgres@localhost/test?encoding=utf8
+      prologue:
+        commands:
+          - checkout
+          - cache restore gems-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),gems-$SEMAPHORE_GIT_BRANCH-,gems-master-
+          - cache restore node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum yarn.lock),node-modules-$SEMAPHORE_GIT_BRANCH,node-modules-master
+          - sem-service start postgres
+          - cd /home/semaphore/.rbenv/plugins/ruby-build && git pull && cd -
+          - sem-version ruby $(cat .ruby-version)
+          - bundle install --deployment --path vendor/bundle
+          - bin/rails db:setup
+      jobs:
+        - name: Unit tests
+          commands:
+            - bin/rails test
+
+        - name: System tests
+          commands:
+            - bin/rails test:system
+
+promotions:
+  - name: Deploy to Heroku staging
+    pipeline_file: heroku-staging.yml
+
```

Comment:
Are you thinking of adding automatic deployments to staging from master? What about production? If yes, we could configure them in this PR as well.

Reply:
Yes, automatic deployments to both staging and production would be good. I will have a look how to do it.

Reply:
I added configuration for auto deployments , but to check it we actually have to merge into master:)